libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -788,11 +800,23 @@ protected TokenEndpointResponse performRequest(TokenEndpointResponse tr) throws
             try {
                 id = OAuth2.callIdentityService(
                     HttpAccess.DEFAULT, tr.idUrlWithInstance, tr.authToken);
-            } catch(Exception e) {
+                isSalesforceIntegrationUser = fetchIsSalesforceIntegrationUser(tr);
```

Comment:
Just checking again, as well, that there's no way to fetch this new attribute in a request that's already part of the flow.  I recall that from the conversation, but it does seem worth triple-checking.

Reply:
I believe that's correct. We need an extra call to get that info.

Reply:
Thanks for confirming, @wmathurin.  🍻 In that case, think that dovetails in `FinishAuthTask` as I have it here? I thought it would since we can capture the new parameter in the background stage and then it's available in the UI/`main` thread stage at the same time as all our other checks.  I was hoping that follows the existing pattern, but there's a lot of code where I might have missed a better home for this logic.

Reply:
I think it belongs there too.

Reply:
Thanks for confirming.  I added just a little documentation and polish in 👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2414/commits/09135e08385d53c86d1a2e16d0d863f5c31b736a